splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsAdmin.java

```diff
@@ -1377,4 +1203,192 @@ private static void dropColumnStatistics(long conglomerateId,
         dd.deleteColumnStatisticsByColumnId(conglomerateId, columnId, tc);
 
     }
+
+    private static class CreateStatisticTask implements Callable<Iterable> {
+
+        private StatisticsOperation statisticsOperation;
+        private boolean mergeStats;
+        private final DataDictionary dataDictionary;
+        private final TransactionController tc;
+        private final HashMap<Long, Pair<String, String>> displayPair;
+        private final ContextManager parent;
+
+        private CreateStatisticTask(StatisticsOperation statisticsOperation, boolean mergeStats, final DataDictionary dataDictionary, final TransactionController tc, final HashMap<Long, Pair<String, String>> displayPair) {
+            this.statisticsOperation = statisticsOperation;
+            this.mergeStats = mergeStats;
+            this.dataDictionary = dataDictionary;
+            this.tc = tc;
+            this.displayPair = displayPair;
+            this.parent = ContextService.getService().getCurrentContextManager();
+        }
+
+        @Override
+        public Iterable call() throws Exception {
+            ContextManager cm = ContextService.getService().newContextManager(parent);
```

Comment:
Every place where newContextManager is currently called has logic to call resetCurrentContextManager and removeContextManager when the operation is complete, but that is not being done here.  Is there a reason?  It may be good to examine if this could cause a resource leak.  Is there some other class that manages the ContextManager for us that we could use?

Reply:
Very good catch, @msirek!

Reply:
See _AbstractBroadcastJoinFlatMapFunction.init()_ for a clean example (_SpliceTransactionManager_ is not an example to follow and should be refactored).

Reply:
I've added the final block to closed ContextManager resources.

Reply:
looks good